Arm/rover_arm_system.hpp

```diff
@@ -3,58 +3,199 @@
+    // elbow
+    units::angular_velocity::revolutions_per_minute_t elbow_speed(
+        static_cast<float>(mc_data.elbow_speed));
+    units::angle::degree_t elbow_angle(static_cast<float>(mc_data.elbow_angle));
+    // wrist
+    units::angular_velocity::revolutions_per_minute_t wrist_speed(
+        static_cast<float>(mc_data.wrist_speed));
+    // units::angle::degree_t
+    // wrist_roll(static_cast<float>(mc_data.wrist_roll)); units::angle::degree_t
+    // wrist_pitch(static_cast<float>(mc_data.wrist_pitch));
+
+    // TODO: implement different arm drive modes in this function.
+
+    // D drive mode
+    rotunda_.SetSpeed(rotunda_speed);
+    rotunda_.SetPosition(rotunda_angle);
+    shoulder_.SetSpeed(shoulder_speed);
+    shoulder_.SetPosition(shoulder_angle);
+    elbow_.SetSpeed(elbow_speed);
+    elbow_.SetPosition(elbow_angle);
+  }
+
+  void HomeArm()
+  {
+    // first getting the accelerometer readings/storing readings
+    Accelerometer::Acceleration_t rotunda_acceleration =
+        rotunda_.GetAccelerometerData();
+    Accelerometer::Acceleration_t shoulder_acceleration =
+        shoulder_.GetAccelerometerData();
+    // then homing the different parts of the arm
+    HomeShoulder(rotunda_acceleration, shoulder_acceleration);
+
+    Accelerometer::Acceleration_t elbow_acceleration =
+        elbow_.GetAccelerometerData();
+    HomeElbow(rotunda_acceleration, elbow_acceleration);
+
+    HomeWrist();
+  }
+  void HomeShoulder(Accelerometer::Acceleration_t rotunda_acceleration,
+                    Accelerometer::Acceleration_t shoulder_acceleration)
+  {
+    double gravity                = 9.81;
+    double rotunda_acceleration_x = static_cast<double>(rotunda_acceleration.x);
+    double rotunda_acceleration_y = static_cast<double>(rotunda_acceleration.y);
+    double shoulder_acceleration_x =
+        static_cast<double>(shoulder_acceleration.x);
+    double shoulder_acceleration_y =
+        static_cast<double>(shoulder_acceleration.y);
+    if (rotunda_acceleration_x == 0.0)
+    {  // catches division by 0
+      rotunda_acceleration_x = .0001;
+    }
+    if (rotunda_acceleration_y == 0.0)
+    {  // catches division by 0
+      rotunda_acceleration_y = .0001;
+    }
+    if (shoulder_acceleration_x == 0.0)
+    {  // catches division by 0
+      shoulder_acceleration_x = .0001;
+    }
+    if (shoulder_acceleration_y == 0.0)
+    {  // catches division by 0
+      shoulder_acceleration_y = .0001;
+    }
+    double rotunda_angle =
+        acos(gravity / rotunda_acceleration_y) *  // for angle
+        (rotunda_acceleration_x /
+         fabs(rotunda_acceleration_x));  // for direction
+    double shoulder_angle =
+        acos(gravity / shoulder_acceleration_y) *  // for angle
+        (shoulder_acceleration_x /
+         fabs(shoulder_acceleration_x));  // for direction
+    units::angle::degree_t home =
+        static_cast<units::angle::degree_t>(rotunda_angle + shoulder_angle);
```

Comment:
We should build out functions for each of these calculation that way we can more easily test the logic and also cleanup code base.